src/Tasks/ResolveManifestFiles.cs

```diff
@@ -855,6 +885,56 @@ IEnumerator IEnumerable.GetEnumerator()
         }
         #endregion
 
+        #region SatelliteRefAssemblyMap
+        private class SatelliteRefAssemblyMap : IEnumerable
+        {
+            private readonly Dictionary<string, MapEntry> _dictionary = new Dictionary<string, MapEntry>();
+
+            public MapEntry this[string fusionName]
+            {
+                get
+                {
+                    string key = fusionName.ToLowerInvariant();
+                    _dictionary.TryGetValue(key, out MapEntry entry);
+                    return entry;
+                }
+            }
+
+            public bool ContainsItem(ITaskItem item)
+            {
+                AssemblyIdentity identity = AssemblyIdentity.FromManagedAssembly(item.ItemSpec);
+                if (identity != null)
+                {
+                    string key = identity.ToString().ToLowerInvariant();
+                    return _dictionary.ContainsKey(key);
+                }
+                return false;
+            }
+
+            public void Add(ITaskItem item)
+            {
+                var entry = new MapEntry(item, true);
+                AssemblyIdentity identity = AssemblyIdentity.FromManagedAssembly(item.ItemSpec);
+                if (identity != null && !String.Equals(identity.Culture, "neutral", StringComparison.Ordinal))
+                {
+                    // Use satellite assembly strong name signature as key
+                    string key = identity.ToString().ToLowerInvariant();
```

Comment:
Instead of having to create this new string, can you use an invariant comparer on the dictionary?

Reply:
changed per suggestion.